### PR TITLE
LPS-195458: A generic date facet

### PR DESCRIPTION
**Pull request by peerkar**
- Story: https://liferay.atlassian.net/browse/LPS-195458
- Covered subtasks:
  - https://liferay.atlassian.net/browse/LPS-195459
  - https://liferay.atlassian.net/browse/LPS-195460
  - https://liferay.atlassian.net/browse/LPS-195895

### CODE REVIEW

**Comment by peerkar** [comment]
ci:test:sf

**Comment by peerkar** [comment]
ci:test:sf

**Comment by peerkar** [comment]
ci:test:relevant

**Comment by peerkar** [comment]
ci:test:sf

**Comment by peerkar** [comment]
ci:test:relevant

**Comment by peerkar** [comment]
ci:test:sf

**Comment by peerkar** [comment]
ci:test:relevant

**Comment by joshchong** [comment]
ci:test:sf

**Comment by joshchong** [comment]
ci:test:search

**Comment by joshchong** [comment]
ci:test:search-solr

**Comment by joshchong** [comment]
the SF failure is unrelated and being addressed here: https://liferay.slack.com/archives/C1SJ64S9X/p1695033165091759

File: modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/date/facet/portlet/DateFacetPortlet.java [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+
+		renderRequest.setAttribute(
+			WebKeys.PORTLET_DISPLAY_CONTEXT, dateFacetDisplayContext);
+
+		super.render(renderRequest, renderResponse);
+	}
+
+	@Activate
+	protected void activate(Map<String, Object> properties) {
+		if (FeatureFlagManagerUtil.isEnabled("LPS-153839")) {
```
Comment (BryanEngler) [inline modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/date/facet/portlet/DateFacetPortlet.java:98]: when i set the FF to true im getting some kind of infinite loop here... the activate method is continually called, maybe triggered by configuration.update()? and the widget doesnt appear in the "Add" panel, so i cant test it on a page. did you see anything like this with your testing?

Comment (BryanEngler) [reply modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/date/facet/portlet/DateFacetPortlet.java:98]: worked around this setting FF to false and adding directly to category.search, but when saving the widget configuration i get `java.lang.IllegalArgumentException: Invalid range syntax [past-hour TO *]`

**Comment by joshchong** [comment]
@BryanEngler, I wrote some of my findings based on the test results in the QA slack thread but thought I'd share it here as well:

There are two failures that look related to your changes:

1. The [com.liferay.portlet.title.localization.test.PortletTitleTest > testPortletTitles](https://testray.liferay.com/home/-/testray/case_results/2426738250) test is failing with: `Please update the Language.properties files for the following portlets: [com_liferay_portal_search_web_date_facet_portlet_DateFacetPortlet]`
2. When navigating to Product Menu > Design > Templates > Widget Templates, the following error is thrown in the console:
```2023-09-18 22:39:24.678 ERROR [http-nio-8080-exec-1][AdvancedPermissionChecker:1276] null
com.liferay.portal.kernel.exception.NoSuchResourceActionException: com_liferay_portal_search_web_date_facet_portlet_DateFacetPortlet#ADD_PORTLET_DISPLAY_TEMPLATE
```

This is causing the [SearchUpgrade#ViewSearchWidgetTemplatesArchive7413U43](https://testray.liferay.com/home/-/testray/case_results/2426737805) test to fail.

**Comment by BryanEngler** [comment]
@peerkar I have some WIP here if you want to take a look https://github.com/BryanEngler/liferay-portal/commits/2023_09_18_pr1072_liferay_search_LPS_195459_Generic_Date_Facet

File: modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/date/facet/portlet/DateFacetPortlet.java (hunk 1 above)
Comment (peerkar) [reply modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/date/facet/portlet/DateFacetPortlet.java:98]: Interesting, didn't have issues with the activate(). @joshchong ?

The validation issue was a copy&paste issue. Fixed.

**Comment by peerkar** [comment]
@BryanEngler glad you corrected those already existing issues with aggregation keys etc. 👍  Didn't mind to touch them.

**Comment by peerkar** [comment]
Resent here: https://github.com/liferay-search/liferay-portal/pull/1080

**Pull request closed without merging**